### ISSUE TITLE
Knight Rider is wrongly marked as watched in Kodi

### ISSUE THREAD

**Issue by JonasL1** [razzeee/script.trakt#375]
Since Kodi 17 my episodes from Knight Rider are marked as watched in Kodi, but they will not be synced as watched to Trakt. I have watched Knight Rider (2008) and not Knight Rider. In the log from Kodi he finds Knight Rider in my collection with the right tvdb ID on line 39. But then on line 200-202 when the watched episodes from Trakt are synced to Kodi, one show is missing playcounts. Then Knight Rider with the wrong tvdb ID (the one from 2008 version) is taken but my corresponding episodes from Knight Rider are marked as watched. This wasn't happening before Kodi 17 and Knight Rider (2008) is no longer in my Kodi library.

I don't know if this bug is related to Trakt or Kodi, but a possible solution could be to compare the tvdb ID's from the Kodi library and the one that are later taken from Trakt so that only the shows with matching ID's are synced.

The link provided should be copied and pasted in the browser, otherwise it would not open.
[https://drive.google.com/file/d/0B3SoZkANly9aUGZnaDNnaEhXZU0/view](url)

**Comment by sharkykh** [comment]
A similar thing just happend to me with the "Power Rangers" movie.
The [2017 feature film](https://trakt.tv/movies/power-rangers-2017) is not watched in Trakt,
and the [2015 short film](https://trakt.tv/movies/power-rangers-2015) is watched.
I only have Power Rangers 2017 in my library.
When syncing, it gets wrongfully marked as watched in Kodi.
By the way, this happens regardless of the value of this setting:
![image](https://user-images.githubusercontent.com/10238474/27264941-ba6f152c-5493-11e7-9f37-b57b80d82227.png)
Also, **I don't think it happens with `togglewatched` action**,
because when I tried that it didn't change anything on Trakt.

@JonasL1 
**To get around this I manually changed the movie title in Kodi to "Power Rangers (2017)".**
And another BTW, on Github you can just paste urls and they because links automatically.
`http://www.google.com/` or you can give the link a label `[label](link)`
https://help.github.com/articles/basic-writing-and-formatting-syntax/#links

**Comment by razzeee** [comment]
Yes, if your scraping the movie wrong, we will send the wrong movie to trakt. I would recommend to attach years to movies and tvshows in every instance.

**Comment by sharkykh** [comment]
I use TMDB, so I'm not sure what I'm doing wrong. I think Kodi has the year stored in the movie information, just not in the title, but I guess it doesn't work like that.

**Comment by razzeee** [comment]
Kodi takes whatever you give it. If you give it the name and the year in the filename it will improve kodis data.

**Comment by JonasL1** [comment]
I named my series liked they are named on thetvdb where Kodi gets the data from. But I can try to add the year and see if the problem is fixed. Strange thing is like I said before that first the right ID is found but later the wrong ID is used.

**Comment by jarlave** [comment]
got the same problem with macgyver 2016 now, in my opinion the ID should be used and not the series title to sync the watched state, I scrape all my shows local in NFO files, so the IDs are all right, title search will never 100% accurate

**Comment by razzeee** [comment]
Well we're using ids, if you provide them. This is the logic if you provide all you can provide:

> Media objects will be matched by ID first, then fall back to title and year. IDs will be matched in this order trakt, imdb, tmdb, tvdb, and slug. If nothing is found, it will match on the title and year. If still nothing, it would use just the title (or name for people) and find the most current object that exists.

**Comment by ckeech** [comment]
@Razzeee When you say "Media objects will be matched by ID first" I'm a bit confused. If a movie or tv show has been scraped correctly in Kodi, will this script then use the ID instead of falling back to other data?

I raised: https://github.com/trakt/script.trakt/issues/374 then found this issue here which seems to reference a similar issue to what I was having. My movies have been correctly scraped so it would seem like they should be passing through IMDB or moviedb ID's to trakt but somewhere it seems to mark 2 movies with the same name different year as watched. This seems like it's falling back on something, but not even title and year seems to be working.
